### ISSUE TITLE
billiard.exceptions.WorkerLostError: Worker exited prematurely: signal 9 (SIGKILL) with billiard pool

### ISSUE THREAD

**Issue by veerendrapolepalli** [celery/billiard#319]
It happening for me with billiyard only , i have tried running it from the command line with or with out using celery ,

File "/venv7/lib/python3.7/site-packages/billiard/pool.py", line 1399, in starmap
starmapstar, chunksize).get()
File "/venv7/lib/python3.7/site-packages/billiard/pool.py", line 1791, in get
raise self._value.exception
File "/venv7/lib/python3.7/site-packages/billiard/pool.py", line 1267, in mark_as_worker_lost
human_status(exitcode)),
billiard.exceptions.WorkerLostError: Worker exited prematurely: signal 9 (SIGKILL).

i am using celery latest version 4.4.7 and billiard 3.6.3.0 and kombu 5.0.1

When i was trying to run the multiprocessing pool job in celery it wasnt not allowing to run it , so i shifted billiard pool for parallel processing

i was trying to run about from command line with 32 parallel jobs using billiard i got this error and if shift back to multiprocessing i am not getting this error.

Stuck with this problem in celery , please suggest me if i need to change any default settings

**Comment by h0rn3t** [comment]
same  problem 

```
[2021-09-28 18:18:59,322: ERROR/MainProcess] Process 'ForkPoolWorker-3' pid:10 exited with 'signal 9 (SIGKILL)'
[2021-09-28 18:18:59,436: ERROR/MainProcess] Task handler raised error: WorkerLostError('Worker exited prematurely: signal 9 (SIGKILL) Job: 0.')
Traceback (most recent call last):
  File "/home/app/.local/lib/python3.9/site-packages/billiard/pool.py", line 1265, in mark_as_worker_lost
    raise WorkerLostError(
billiard.exceptions.WorkerLostError: Worker exited prematurely: signal 9 (SIGKILL) Job: 0.

```

**Comment by auvipy** [comment]
you should try celery latest and report again

**Comment by alvaro-scelza** [comment]
Hi @veerendrapolepalli, @h0rn3t! Have any of you guys found a solution to this problem?
I'm having the same error, using Celery current last version `celery[auth,redis,tblib]==5.2.6` and `billiard==3.6.4.0`.

**Comment by ArslanDevPy** [comment]
**celery==5.2.7
billiard==3.6.4.0**
[2022-11-08 10:22:12,972: ERROR/MainProcess] Process 'ForkPoolWorker-4' pid:19 exited with 'signal 9 (SIGKILL)'
[2022-11-08 10:22:13,375: ERROR/MainProcess] Task handler raised error: WorkerLostError('Worker exited prematurely: signal 9 (SIGKILL) Job: 10.')
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/billiard/pool.py", line 1265, in mark_as_worker_lost
    raise WorkerLostError(
billiard.exceptions.WorkerLostError: Worker exited prematurely: signal 9 (SIGKILL) Job: 10.

**Comment by SchnitzeI** [comment]
I am experiencing the same problem, some PDFs cannot be processed by ocr (celery)

**Comment by auvipy** [comment]
I found this post and issue might be related https://github.com/celery/celery/issues/7324#issuecomment-1751302840 can you check please?

**Comment by SchnitzeI** [comment]
tl;dr increasing the machines memory fixed the issue for me 👍 
there is also a possible solution to deactivate searching for barcodes if u are short on memory - with environment variable PAPERLESS_CONSUMER_ENABLE_ASN_BARCODE=false

Hi @auvipy, thank you for the fast reply. I tried to figure it out, but couldnt find the celery service file for configuration. Randomly I saw my machine was eating all the memory at once so I increased it, now it works flawless.

Have a great day :)
